commit 88d253ea23b06289df40401160b606323f16c910
Author: Song Bing <b06498@freescale.com>
Date:   Mon Dec 15 09:34:35 2014 +0800

videopool: update video alignment after video alignment

Video buffer pool will update video alignment to respect stride alignment
requirement. But haven't update it to video alignment in configure.
Which will cause user get wrong video alignment.

https://bugzilla.gnome.org/show_bug.cgi?id=741501

Upstream-Status: Backport [1.5.1]

Signed-off-by: Song Bing <b06498@freescale.com>
diff --git a/gst-libs/gst/video/gstvideopool.c b/gst-libs/gst/video/gstvideopool.c
index 4475f45..acef594 100644
--- a/gst-libs/gst/video/gstvideopool.c
+++ b/gst-libs/gst/video/gstvideopool.c
@@ -167,6 +167,7 @@ video_buffer_pool_set_config (GstBufferPool * pool, GstStructure * config)
     /* get an apply the alignment to the info */
     gst_buffer_pool_config_get_video_alignment (config, &priv->video_align);
     gst_video_info_align (&info, &priv->video_align);
+    gst_buffer_pool_config_set_video_alignment (config, &priv->video_align);
   }
   priv->info = info;

diff --git a/sys/ximage/ximagepool.c b/sys/ximage/ximagepool.c
index 6cc2cfa..6a1cbc9 100644
--- a/sys/ximage/ximagepool.c
+++ b/sys/ximage/ximagepool.c
@@ -597,6 +597,8 @@ ximage_buffer_pool_set_config (GstBufferPool * pool, GstStructure * config)
     /* do padding and alignment */
     gst_video_info_align (&info, &priv->align);

+    gst_buffer_pool_config_set_video_alignment (config, &priv->align);
+
     /* we need the video metadata too now */
     priv->add_metavideo = TRUE;
   } else {
diff --git a/sys/xvimage/xvimagepool.c b/sys/xvimage/xvimagepool.c
index 244a51a..34b1ab2 100644
--- a/sys/xvimage/xvimagepool.c
+++ b/sys/xvimage/xvimagepool.c
@@ -124,6 +124,8 @@ xvimage_buffer_pool_set_config (GstBufferPool * pool, GstStructure * config)
     /* do padding and alignment */
     gst_video_info_align (&info, &priv->align);

+    gst_buffer_pool_config_set_video_alignment (config, &priv->align);
+
     /* we need the video metadata too now */
     priv->add_metavideo = TRUE;
   } else {
